src/Elmish.WPF/WpfProgram.fs

```diff
@@ -0,0 +1,137 @@
+﻿namespace Elmish.WPF
+
+open System.Windows
+open Microsoft.Extensions.Logging
+open Microsoft.Extensions.Logging.Abstractions
+open Elmish
+
+
+type WpfProgram<'model, 'msg> =
+  internal {
+    ElmishProgram: Program<unit, 'model, 'msg, Binding<'model, 'msg> list>
+    LoggerFactory: ILoggerFactory
+    /// Only log calls that take at least this many milliseconds. Default 1.
+    PerformanceLogThreshold: int
+  }
+
+
+[<RequireQualifiedAccess>]
+module WpfProgram =
+
+
+  let private fromElmishProgram program =
+    { ElmishProgram = program
+      LoggerFactory = NullLoggerFactory.Instance
+      PerformanceLogThreshold = 1 }
+
+
+  /// Creates a WpfProgram that does not use commands.
+  let mkSimple
+      (init: unit -> 'model)
+      (update: 'msg  -> 'model -> 'model)
+      (bindings: unit -> Binding<'model, 'msg> list) =
+    Program.mkSimple init update (fun _ _ -> bindings ())
+    |> fromElmishProgram
+
+
+  /// Creates a WpfProgram that uses commands
+  let mkProgram
+      (init: unit -> 'model * Cmd<'msg>)
+      (update: 'msg  -> 'model -> 'model * Cmd<'msg>)
+      (bindings: unit -> Binding<'model, 'msg> list) =
+    Program.mkProgram init update (fun _ _ -> bindings ())
+    |> fromElmishProgram
```

Comment:
Now that Elmish.WPF has its own `Program` type, can store the `bindings` there (and get them there by passing them into `fromElmishProgram`) instead passing them through the Elmish program in this obscure way?

Reply:
Not sure I understand, but we need a "view" to be able to create a `Program`. We could defer the `Program` creation until `startElmishLoop`, but then we also need to keep track of all `Program` modifications, such as `withSubscription`. I'm not convinced that is more clear than simply "extending" the existing `Program` like we have done.

Reply:
I specifically mean the change in [this commit](https://github.com/bender2k14/Elmish.WPF/commit/febe5bcd44fe277ca8ebcc6b49a206608e251962) that I just made.  This code makes it easier to understand where the bindings come from.  The current code adds this unnecessary layer of direction via the `View` function.

Reply:
Oh, I see. It's about the `let bindings = Program.view ...` line where we extract the bindings again. Sure, I'll change it. Thanks!